# testing the accuracy of the conensus sequences generated by medaka to known orthologs of from 427


#mafft --auto reordered_fastas/concatenated_sequences.fasta > reordered_fastas/aligned_genomes.fasta

#cat 427_Orthologs/Tb927.1.3230 Consensus_genes/Tb927.1.3230.fasta > 427_Orthologs/Inputs/input.fasta
#mafft --auto 427_Orthologs/Inputs/input.fasta > 427_Orthologs/Test_aligns/Tb927.1.3230_Test



for sample in /Volumes/Seagate/monotrack/427_Orthologs/Tb927*
 
    do
    base=$(basename "$sample")
    dir="/Volumes/Seagate/monotrack/427_Orthologs"
    input_dir="$dir/Inputs"
    align_dir="$dir/Test_aligns"

    cat $dir/${base} $dir/Consensus_genes/${base}.fasta > $input_dir/${base}.input.fasta
    mafft --auto $input_dir/${base}.input.fasta > $align_dir/${base}.aligned 
done